jujugui/static/gui/src/app/components/terminal/test-terminal.js

```diff
@@ -77,6 +79,29 @@ describe('Terminal', () => {
     assert.equal(websocket.prototype.close.callCount, 1);
   });
 
+  it('sends supplied commands when it is set up', () => {
+    setupWebsocket();
+    const component = ReactTestUtils.renderIntoDocument(
+      <Terminal
+        addNotification={sinon.stub()}
+        address="1.2.3.4:123"
+        changeState={sinon.stub()}
+        commands={['juju status']}
+        creds={{
+          user: 'user',
+          password: 'password',
+          macaroons: {}
+        }}
+        WebSocket={websocket} />
+    );
+    // Send the setup from the term.
+    component.ws.onmessage({data: '["setup", {}]'});
+    // Send the initial PS1
+    component.ws.onmessage({data: '["stdout", "\\u001b[01;32"]'});
+    ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(component).parentNode);
+    assert.equal(websocket.prototype.send.callCount, 1);
```

Comment:
Please also test what's actually sent, and also test with multiple commands.

Reply:
done